cmd/fleet/serve.go

```diff
@@ -631,6 +632,29 @@ the way that the Fleet server works.
 				if appCfg.MDM.EnabledAndConfigured {
 					profileMatcher = apple_mdm.NewProfileMatcher(redisPool)
 				}
+				var softwareInstallStore fleet.SoftwareInstallerStore
+				if config.S3.Bucket != "" {
+					store, err := s3.NewSoftwareInstallerStore(config.S3)
+					if err != nil {
+						initFatal(err, "initializing S3 software installer store")
+					}
+					softwareInstallStore = store
+					level.Info(logger).Log("msg", "using S3 software installer store", "bucket", config.S3.Bucket)
+				} else {
+					// TODO(mna): to clarify which directory we use: https://github.com/fleetdm/fleet/issues/18329#issuecomment-2072303528
```

Comment:
This still needs resolution, though maybe we're good now with the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` env var that we can set to use something else than tempdir, and otherwise use that?

Reply:
I see that `os.TempDir` uses `$TMPDIR` by default, in my macOS machine is inside `/var`:

```
$ echo $TMPDIR
/var/folders/t5/nbc8hjfn5yq1qv8s26ngq07c0000gn/T/
```

so it might be a good argument to just use it? specially considering that `fleetctl preview` is the only place where this is supposed to work

Reply:
Yeah my guess is that TempDir with an option to override it with the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` env var is good enough.

Reply:
I'll remove the TODO.